### PR TITLE
add Exists method

### PR DESCRIPTION
**Pull request by natebrennand**
Also switched to `gomock`...

### CODE REVIEW

File: pathio.go [hunk 1]
```diff
@@ -114,6 +112,41 @@ func (c *Client) ListFiles(path string) ([]string, error) {
...
+	if err != nil {
+		if aerr, ok := err.(s3.RequestFailure); ok && aerr.StatusCode() == 404 {
+			return false, nil
+		}
+		return false, err
+	}
+	return true, nil
+}
+
+func existsLocal(path string) (bool, error) {
```
Comment (potluck) [inline pathio.go]: @natebrennand just curious - is this intentionally not being called in `Exists`?

Comment (natebrennand) [inline pathio.go]: Nope, that's just a bug 😕
Will fix tomorrow, nice catch

Comment (natebrennand) [inline pathio.go]: I think I put the default response in `Exists` to get the linter to shut up will I was implementing `existsLocal`

Comment (potluck) [inline pathio.go]: makes sense!

**Pull request merged**